Description: mod_pubsub: Don't blindly echo PEP notification
Author: Holger Weiss <holger@zedat.fu-berlin.de>
Origin: upstream
Bug: https://github.com/processone/ejabberd/issues/4158
Applied-Upstream: 3469a51f5896ef96b2e53d61fcfd2163b92ad11a
Last-Update: 2024-09-15
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/mod_pubsub.erl
+++ b/src/mod_pubsub.erl
@@ -3051,8 +3051,7 @@
 	       extended_headers([Publisher])),
     Pred = fun(To) -> delivery_permitted(Owner, To, NodeOptions) end,
     ejabberd_sm:route(jid:make(LUser, LServer, SenderResource),
-		      {pep_message, <<((Node))/binary, "+notify">>, Stanza, Pred}),
-    ejabberd_router:route(xmpp:set_to(Stanza, jid:make(LUser, LServer)));
+		      {pep_message, <<((Node))/binary, "+notify">>, Stanza, Pred});
 broadcast_stanza(Host, _Publisher, Node, Nidx, Type, NodeOptions, SubsByDepth, NotifyType, BaseStanza, SHIM) ->
     broadcast_stanza(Host, Node, Nidx, Type, NodeOptions, SubsByDepth, NotifyType, BaseStanza, SHIM).
 
